### ISSUE TITLE
extends \Redis problem

### ISSUE THREAD

**Issue by Yurunsoft** [phpredis/phpredis#1386]
### I'm seeing this behaviour on
- OS: Ubuntu
- Redis: 3.2.100
- PHP: 7.1.20
- phpredis: 4.0.0

### Steps to reproduce, backtrace or example script

```php
<?php
class MyRedis extends \Redis
{
	public function set($key,$value,$timeout = null)
	{
		parent::set($key, $value, $timeout);
	}
}
$r1 = new MyRedis;
$r1->connect('192.168.0.222', 6379);
$r1->set(1,2);

echo PHP_EOL, '-----', PHP_EOL, PHP_EOL;

class MyRedis2 extends \Redis
{
	public function set($key, $value, $timeout = null, $opt = null)
	{
		parent::set($key, $value, $timeout, $opt);
	}
}
$r2 = new MyRedis2;
$r2->connect('192.168.0.222', 6379);
$r2->set(1,2);
```

result：

```
PHP Warning:  Declaration of MyRedis::set($key, $value, $timeout = NULL) should be compatible with Redis::set($key, $value, $timeout = NULL, $opt = NULL) in /mnt/d/projects/xxx/test.php on line 8

-----

PHP Warning:  Redis::set() expects at most 3 parameters, 4 given in /mnt/d/projects/xxx/test.php on line 19
```

### I've checked
- [x] There is no similar issue from other users
- [ ] Issue isn't fixed in `develop` branch

**Comment by yatsukhnenko** [comment]
@Yurunsoft this issue already fixed in `4.1.1` and `develop` branch

**Comment by Yurunsoft** [comment]
OK
